libsnark/zk_proof_systems/plonk/prover.hpp

```diff
@@ -0,0 +1,252 @@
+/** @file
+*****************************************************************************
+Declaration of Prover interfaces for ppzkSNARK proof system Plonk.
+
+This includes:
+- class for proof
+- class for prover
+
+References:
+
+\[GWC19]:
+"Plonk: Permutations over lagrange-bases for oecumenical noninteractive
+arguments of knowledge", Ariel Gabizon, Zachary J. Williamson, and Oana
+Ciobotaru, Cryptology ePrint Archive, Report 2019/953, 2019,
+<https://eprint.iacr.org/2019/953>
+
+*****************************************************************************
+* @author     This file is part of libsnark, developed by SCIPR Lab
+*             and contributors (see AUTHORS).
+* @copyright  MIT license (see LICENSE file)
+*****************************************************************************/
+
+#ifndef PLONK_PPZKSNARK_PROVER_HPP_
+#define PLONK_PPZKSNARK_PROVER_HPP_
+
+namespace libsnark
+{
+/**
+ * A proof for the Plonk GG-ppzkSNARK.
+ */
+template<typename ppT> class plonk_proof
+{
+public:
+    using Field = libff::Fr<ppT>;
+
+    // Plonk proof Pi
+    //
+    // Pi ([a]_1, [b]_1, [c]_1, [z]_1,
+    //     [t_lo]_1, [t_mi]_1, [t_hi]_1,
+    //     \bar{a}, \bar{b}, \bar{c},
+    //     \bar{S_sigma1}, \bar{S_sigma2}, \bar{z_w},
+    //     [W_zeta]_1, [W_{zeta omega_roots}]_1
+    //     r_zeta (*))
+
+    // [a]_1, [b]_1, [c]_1
+    std::vector<libff::G1<ppT>> W_polys_blinded_at_secret_g1;
+    // [z]_1,
+    libff::G1<ppT> z_poly_at_secret_g1;
+    // [t_lo]_1, [t_mi]_1, [t_hi]_1
+    std::vector<libff::G1<ppT>> t_poly_at_secret_g1;
+    // \bar{a}, \bar{b}, \bar{c},
+    Field a_zeta;
+    Field b_zeta;
+    Field c_zeta;
+    // \bar{S_sigma1}, \bar{S_sigma2},
+    Field S_0_zeta;
+    Field S_1_zeta;
+    // \bar{z_w}
+    Field z_poly_xomega_zeta;
+    // [W_zeta]_1, [W_{zeta omega_roots}]_1
+    libff::G1<ppT> W_zeta_at_secret;
+    libff::G1<ppT> W_zeta_omega_at_secret;
+    // r_zeta (*)
+    Field r_zeta;
+
+    plonk_proof(){};
```

Comment:
What is this for?  It seems to just enable the possibility of uninitialized classes?

Reply:
Redundant. Removed.